### PR TITLE
Get rid of non-standard imports and one-line functions

### PR DESCRIPTION
**Pull request by jbrockmendel**
Having to look these up just adds another layer for the reader to sort through.

### CODE REVIEW

File: statsmodels/tsa/vector_ar/vecm.py [hunk 1]
```diff
@@ -609,23 +609,10 @@ def coint_johansen(endog, det_order, k_ar_diff):
                       category=HypothesisTestWarning)
 
     from statsmodels.regression.linear_model import OLS
-    tdiff = np.diff
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/vecm.py]: This saves all of two keystrokes, and is used only once.

File: statsmodels/tsa/vector_ar/vecm.py [hunk 2]
```diff
@@ -609,23 +609,10 @@ def coint_johansen(endog, det_order, k_ar_diff):
...
 
     class Holder(object):
         pass
 
-    def trimr(x, front, end):
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/vecm.py]: This is only ever called with `end=0`, so the function reduces to the one-liner `x[front:]`.  The reader shouldn't have to look this up.

File: statsmodels/tsa/vector_ar/vecm.py [hunk 3]
```diff
@@ -609,23 +609,10 @@ def coint_johansen(endog, det_order, k_ar_diff):
...
-        if end > 0:
-            return x[front:-end]
-        else:
-            return x[front:]
-
-    import statsmodels.tsa.tsatools as tsat
-    mlag = tsat.lagmat
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/vecm.py]: No!

File: statsmodels/tsa/vector_ar/vecm.py [hunk 4]
```diff
@@ -609,23 +609,10 @@ def coint_johansen(endog, det_order, k_ar_diff):
...
-
-    def lag(x, lags):
```
Comment (jbrockmendel) [inline statsmodels/tsa/vector_ar/vecm.py]: This is bad and it should feel bad.

**Comment by josef-pkt** [comment]
http://www.statsmodels.org/devel/dev/index.html?highlight=commit%20message#making-changes-to-the-code
see paragraph 2 on commit messages

**Comment by jbrockmendel** [comment]
OK I see it.  What of it?

**Comment by jbrockmendel** [comment]
@ChadFulton Don't be rated PG; be rated R!

**Comment by ChadFulton** [comment]
> OK I see it. What of it?

I guess Josef is suggesting that you use commit prefixes? I also like them.

**Comment by ChadFulton** [comment]
Aside from that, though, this PR looks fine to me.

**Comment by josef-pkt** [comment]
@ChadFulton 
You should add the missing commit code in the merge message if it is not there already, so we still get the commit codes when listing the merge commits.

**Comment by josef-pkt** [comment]
PR should be rebased before merging. The PR is based on master from April.

**Comment by josef-pkt** [comment]
rebased, edited commit message and force pushed

**Comment by josef-pkt** [comment]
This pull request **fixes 1 alert** when merging cfe9871e76393db7bda1e8f95c45fa8d2ecef7fb into a24d18cc638abfbf2fb2fa92bfac54e8b7f87af7 - [view on LGTM.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-255b864b67d6f1f42e172317e12823f194778070)

**fixed alerts:**

* 1 for Module is imported with &#39;import&#39; and &#39;import from&#39;

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by josef-pkt** [comment]
removing `lag` was not a good idea, makes code more difficult to understand, i.e. I have to think about indices instead of reading a descriptive function name like `lag`

**Comment by jbrockmendel** [comment]
Even if so, we all know that readability doesn't matter.

**Comment by josef-pkt** [comment]
What do the 20 to 50 lines starting in https://github.com/statsmodels/statsmodels/blob/master/statsmodels/tsa/vector_ar/vecm.py#L629 do and is there a bug?

Somebody wrote that code 6 years ago (and the background matlab function is quite a bit older).
Somebody needs to find a bug or justify why there is no bug.

(maybe there is someone for the second somebody that finds the code "readable".)

**Comment by jbrockmendel** [comment]
> What do the 20 to 50 lines starting in https://github.com/statsmodels/statsmodels/blob/master/statsmodels/tsa/vector_ar/vecm.py#L629 do and is there a bug?

Off the top of my head, no idea.  Last time I reviewed vecm it was a bug-filled mess, so probably.

Is there something in particular you're trying to accomplish by complaining about python syntax in a long-closed thread?  Or just hammering home the point "readability counts for my idiosyncratic preferences and everyone else can take a hike"?

**Comment by josef-pkt** [comment]
I'm just making a general statement (which will likely affect future PRs) that one or two liner functions can be useful if the function has a descriptive name that helps in reading code.
(comment by a idiosyncratic bug-hunter)

(I'm currently trying to sell a matrix_sqrt on scipy dev mailing list that is essentially just a few lines of linalg, which are however difficult to remember for non-linalg developers.)

(It's a bit cold and wet in Montreal for taking a hike, so I might look for some cointegration articles or books instead.)

**Pull request merged**